ibu-imager/clusterinfo/clusterinfo.go

```diff
@@ -77,6 +77,34 @@ func (m *InfoClient) CreateClusterInfo(ctx context.Context) (*ClusterInfo, error
 	}, nil
 }
 
+func (m *InfoClient) GetSecretData(ctx context.Context, name, namespace, key string) (string, error) {
+	secret := &corev1.Secret{}
+	if err := m.client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, secret); err != nil {
+		return "", err
+	}
+
+	data, ok := secret.Data[key]
+	if !ok {
+		return "", fmt.Errorf("did not find key %s in Secret %s/%s", key, name, namespace)
+	}
+
+	return string(data), nil
+}
+
+func (m *InfoClient) GetConfigMapData(ctx context.Context, name, namespace, key string) (string, error) {
```

Comment:
Maybe we can have both functions as one? they both seems to be the same. Possible? Maybe using runtime.Object?

Reply:
tried using something like below

```
	var obj runtime.Object

	switch resourceType {
	case "Secret":
		obj = &corev1.Secret{}
	case "ConfigMap":
		obj = &corev1.ConfigMap{}
	default:
		return "", fmt.Errorf("unsupported resource type: %s", resourceType)
	}
```

but cannot use 'obj' (type Object) as the type Object Type does not implement 'Object' as some methods are missing: 
GetNamespace() string 
SetNamespace(namespace string) 
GetName() string
...